src/error.ts

```diff
@@ -744,20 +752,23 @@ function isRecoveringError(err: MongoError) {
     return SDAM_RECOVERING_CODES.has(err.code);
   }
 
-  return /not master or secondary/.test(err.message) || /node is recovering/.test(err.message);
+  return (
+    LEGACY_NOT_PRIMARY_OR_SECONDARY_ERROR_MESSAGE_REGEX.test(err.message) ||
+    /node is recovering/.test(err.message)
```

Comment:
Can we make a constant for this error message as well?

Reply:
I waffled about this initially. Updated